sfcasts/patterns.md

```diff
@@ -1,62 +1,36 @@
-bit about how the game, how the app is set up right now. This basically just sets
-things up and prints the results. It figures out which character you want to be and
-then it actually starts the battle. Um, but most of the work is actually done, is
-done here on this game object, which is this game application. This takes these two
-character objects and it actually does the logic of having them attack each other
-until somebody wins. It also at the bottom contains the three characters we have,
-which are represented by this character object and that character object. You can
-pass in different max health, different base damage that you do and different armor
-levels. So game application with care three characters here, and that battles them up
-up here, and that's basically it. So next, let's get into our first pattern, which is
-going to be this strategy pattern where we make our characters able to cast magical
-spells, which will require making this class a lot more flexible.
+A design pattern is nothing more than a strategy for writing your code when you have a particular problem. If you can start to identify which types of problems are solved by which strategies, you'll walk into situations and immediately know what to do. Learning these design patterns gives you:
+
+A) More tools in your developer toolkit when coding and
+B) A better understanding of core libraries like Symfony, which leverages design patterns frequently.
+
+It's pretty cool to look at core code and say
+
+`Oh, I see what you're doing there. That's the
+decorator pattern.`
+
+There are tons of design patterns and many of them will likely never be useful to you. They just won't ever apply to problems that you are facing in your real work. These design patterns fall into three basic groups. You don't need to memorize these, but it's a nice way to think about the three types of problems that design pattern solve.
+
+The first type is called "creation patterns", and these are all about helping *instantiate* objects. They include the factory pattern, builder pattern, singleton pattern, and others. The second is called "structural patterns". These help you organize things when you have a bunch of objects and you need to identify relationships between them. One example of a relationship would be a parent-child relationship, but there are *many* others. This one can be a little fuzzy, but we'll see *one* structural pattern in this tutorial - the "decorator pattern". The third and final type of patterns are called "behavioral patterns", which help solve problems with how objects communicate with each other, as well as assigning responsibilities between objects. That's a fancy way of saying that behavioral patterns help you design classes with one specific responsibility that can then work together, instead of putting all of that code into one *giant* class. We'll talk about *two* behavioral patterns - the "strategy pattern" and the "observer pattern".
```

Comment:
Float in text on the screen for each design pattern individually with some info about it. Again, it will approximately match what I'm saying, but should have different wording